Src/Base/AMReX_GpuLaunch.H

```diff
@@ -339,6 +342,39 @@ namespace Gpu {
         block; \
     }
 
+#define AMREX_GPU_LAUNCH_HOST_DEVICE_PARALLEL_FOR_RANGE_FLAG(where_to_run,bx1,nc1,block1) \
```

Comment:
In the existing macros, the `i`, `j`, `k`, `n` are macro arguments.  But the new ones are not, and therefore the users will not be able to change them.  If the enclosing scope has a local variable named `i` or `n`.  This will be confusing.  Also, maybe the name can have `4D` in it to be more explicit as we are currently doing.